spring-kafka/src/main/java/org/springframework/kafka/listener/AbstractConsumerSeekAware.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.listener;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.apache.kafka.common.TopicPartition;
+
+import org.springframework.lang.Nullable;
+
+/**
+ * Manages the {@link ConsumerSeekAware.ConsumerSeekCallback} s for the listener. If the
+ * listener subclasses this class, it can easily seek arbitrary topics/partitions without
+ * having to keep track of the callbacks itself.
+ *
+ * @author Gary Russell
+ * @since 2.3
+ *
+ */
+public abstract class AbstractConsumerSeekAware implements ConsumerSeekAware {
+
+	private static final ThreadLocal<ConsumerSeekCallback> callbackForThread = new ThreadLocal<>();
```

Comment:
`static`?
Shouldn't such a thread store be an instance based?

Reply:
Good catch.